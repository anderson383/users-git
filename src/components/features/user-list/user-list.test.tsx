
import {
  fireEvent,
  render,
  screen
} from '@testing-library/react';
import UserList from '.';
import { BrowserRouter } from 'react-router-dom';
import { act } from 'react-dom/test-utils';
import * as ResizeObserverModule from 'resize-observer-polyfill';

(window as any).ResizeObserver = ResizeObserverModule.default;

const mockUser = [
  {
    avatar_url: 'url image',
    login: 'test user',
    id: 'id-test'
  }
]

const navigateMock = jest.fn()

jest.mock('react-router-dom', () => {
  return ({
    ...jest.requireActual('react-router-dom'),
    useParams: () => jest.fn(),
    useNavigate: () => navigateMock
  }
  )
})

const getUsersMock = jest.fn().mockImplementation(() => Promise.resolve(mockUser))
jest.mock('../../../services/config/context', () => ({
  useRepositoryIoc: jest.fn().mockReturnValue({
    container: {
      get: jest.fn().mockReturnValue({
        getUser: jest.fn().mockImplementation(() => Promise.resolve(mockUser[0])),
        getUsers: () => getUsersMock()
      })
    }
  })
}))


describe('Tests in UserList component', () => {
  const setup = () => {
    return render(
      <BrowserRouter>
        <UserList />
      </BrowserRouter>
    );
  };

  test('Should render the component', async () => {
    const {asFragment} = setup();
    
    await act(() => Promise.resolve())
    expect(asFragment()).toMatchSnapshot();
  });

  test('Should render if no charge list', async () => {
    getUsersMock.mockReturnValue(null)
    setup();
    
    await act(() => Promise.resolve())
    expect(screen.getByText('No hay usuarios en base a tu busqueda...')).toMatchSnapshot();
  });

  test('Should search value', async () => {
    setup();

    const buttSearch = screen.getByTestId('search');

    fireEvent.change(screen.getByPlaceholderText('Buscar'), {target: { value: 'user test' }});

    fireEvent.click(buttSearch);
    await act(() => Promise.resolve())

    expect(navigateMock).toHaveBeenCalled();
  });
});


