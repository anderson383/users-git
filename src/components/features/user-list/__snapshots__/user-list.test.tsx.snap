// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in UserList component Should render if no charge list 1`] = `
<h2>
  No hay usuarios en base a tu busqueda...
</h2>
`;

exports[`Tests in UserList component Should render the component 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h1>
      Listado de usuarios
    </h1>
    <form>
      <div
        class=""
      >
        <div
          class="content"
        >
          <button
            data-testid="search"
            type="submit"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-magnifying-glass "
              data-icon="magnifying-glass"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
                fill="currentColor"
              />
            </svg>
          </button>
          <input
            class="input inputSearch"
            name="search"
            placeholder="Buscar"
            type="text"
            value=""
          />
        </div>
      </div>
    </form>
    <div
      class="dib"
    >
      <canvas
        height="0"
        role="img"
        style="display: block; box-sizing: border-box; height: 0px; width: 0px;"
        width="0"
      />
      ;
    </div>
    <div
      class="container_users"
    >
      <div
        class="card_user"
        data-testid="card_user"
      >
        <div
          class="image"
        >
          <img
            alt=""
            src="url image"
          />
        </div>
        <div
          class="info"
        >
          <p>
            test user
          </p>
          <span>
            id-test
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
